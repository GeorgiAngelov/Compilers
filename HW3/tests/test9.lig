function f(a : int, b : bool, c : int) { }

function main(args : [[int]]) : int {
      f(1, true, 3);
      f(1);
      f(1, true);
      f(1, true, true);
      f();

      return (main());

      // Returning something with a void type.
      return (f(1, true, 3));
}
