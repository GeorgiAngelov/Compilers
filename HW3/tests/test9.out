(decls
   (decl-fun:ok (id f) (type-fun (params (typed-id (id a) (type-int)) (typed-id (id b) (type-bool)) (typed-id (id c) (type-int))) (type-void)))
   (decl-fun:NONE (id main) (type-fun (params (typed-id (id args) (type-array (type-array (type-int))))) (type-int))
      (decls)
      (stmts
         (exp-stmt:ok (fun-call:void (id f) (exps (num:int 1) (true:bool) (num:int 3))))
         (exp-stmt:NONE (fun-call:NONE (id f) (exps (num:int 1))))
         (exp-stmt:NONE (fun-call:NONE (id f) (exps (num:int 1) (true:bool))))
         (exp-stmt:NONE (fun-call:NONE (id f) (exps (num:int 1) (true:bool) (true:bool))))
         (exp-stmt:NONE (fun-call:NONE (id f) (exps)))
         (return:NONE (fun-call:NONE (id main) (exps)))
         (return:NONE (fun-call:void (id f) (exps (num:int 1) (true:bool) (num:int 3)))))))
Typechecking failed!
